snap/snapcraft.yaml

```diff
@@ -68,72 +72,102 @@ parts:
         - libsodium23
         - libxml2
         - libxslt1.1
+        - libpython3.8-minimal
+        - libpython3.8-stdlib
+        - python3-distutils
+        - python3-pip
+        - python3-setuptools
+        - python3.8
         - squashfs-tools
         - xdelta3
+    build-environment:
+        - "CRYPTOGRAPHY_DONT_BUILD_RUST": "1"
     override-build: |
-      snapcraftctl build
+        snapcraftctl build
+
+        echo "Create libsodium symlink..."
+        TRIPLET_PATH="$SNAPCRAFT_PART_INSTALL/usr/lib/$(gcc -print-multiarch)"
+        LIBSODIUM="$(readlink -n "$TRIPLET_PATH/libsodium.so.23")"
+        ln -sf "$LIBSODIUM" "$TRIPLET_PATH/libsodium.so"
+
+        # Default to python3.8.
+        ln -sf python3.8 "$SNAPCRAFT_PART_INSTALL/usr/bin/python3"
 
-      echo "Create libsodium symlink..."
-      TRIPLET_PATH="$SNAPCRAFT_PART_INSTALL/usr/lib/$(gcc -print-multiarch)"
-      LIBSODIUM="$(readlink -n "$TRIPLET_PATH/libsodium.so.23")"
-      # Remove so the link can be recreated on re-builds
-      rm -f "$TRIPLET_PATH/libsodium.so"
-      ln -s "$LIBSODIUM" "$TRIPLET_PATH/libsodium.so"
+        # Site customization to load correct paths depending on running as snap or building.
+        cp "$SNAPCRAFT_PART_SRC/sitecustomize.py" "$SNAPCRAFT_PART_INSTALL/usr/lib/python3.8/sitecustomize.py"
```

Comment:
If you remove it from stage, then this comment makes little sense to me. How does this affect the plugin included sitecustomize?

Reply:
because it _was_ installed/staged in snapcraft-libs (not a python plugin), the sitecustomize wouldn't matter in the subsequent parts (it wouldn't be read unless in $SNAPCRAFT_STAGE).

I dropped this approach for now and stage it in each part.